### PR TITLE
fix(developer): test-page model loading

### PR DESCRIPTION
**Pull request by jahorton**
It would appear that ever since KeymanWeb got modularized, the Developer test-page hooks for loading models have been broken.  With #8472, `keyman.modelManager` was reworked into `keyman.modelCache`. 

The prediction and correction toggle properties... weren't even available there in 16.0, so the Web test-page toggles have likely been dead for a little while.  I went ahead and patched them up while I was at it.

## User Testing

TEST_PREDICTIVE_TOGGLES:  Using the "Prediction - robust testing" for **_KeymanWeb_** **on a mobile device**, verify that the toggles have the indicated effect.

1. Click on a text area.
2. Verify that the predictive-text banner is visible.
3. Click on a blank part of the page (to dismiss the OSK)
4. Click the "Disable predictions" button.
5. Click on the original text area (from step 1)
6. Verify that the predictive-text banner is _not_ visible.
7. Click on a blank part of the page (to dismiss the OSK)
8. Click the "Enable predictions" button.
9. Click on the original text area (from step 1)
10. Verify that the predictive-text banner is visible.

TEST_DEV_HOST_MODELS:  Using the **_Developer_** test-host page, verify that lexical models load properly.
- `release/sil/sil_euro_latin` (for keyboard) and `release/nrc/nrc_en_mtnt` (for model) are probably "good enough" to use for this test.  Feel free to use and/or test with others if you prefer.

1. Select a mobile test device configuration.  (e.g: Device > Phone Devices > "iPhone 5S")
2. Verify that selecting a keyboard (e.g: `sil_euro_latin`) and model (e.g: `nrc_en_mtnt`) with matching language codes causes the predictive-text banner to appear.
    - You may need to open both projects, build them, and request to test them via the project "build" tab for them to appear in the test-host page.
4. Check that interactions in the web debugger continue to work as before.

### CODE REVIEW

**Comment by jahorton** [comment]
I discovered the underlying issue due to a repro attempt (against `master`) for https://community.software.sil.org/t/word-order-in-lexical-models/8268.

**Comment by jahorton** [comment]
iOS failure: 

```
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/spaceship/lib/spaceship/connect_api/api_client.rb:250:in `handle_error': [!] 
    The request could not be completed because: (Spaceship::ProgramLicenseAgreementUpdated)

...

/Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/spaceship/lib/spaceship/connect_api/api_client.rb:250:in `handle_error': 
    A required agreement is missing or has expired. - This request requires an in-effect agreement that has not been signed or has expired. (Spaceship::ProgramLicenseAgreementUpdated)
```

**Comment by mcdurdin** [comment]
> With #8472, keyman.modelManager was reworked into keyman.modelCache.

Was this a public API?

**Comment by jahorton** [comment]
> > With #8472, keyman.modelManager was reworked into keyman.modelCache.
> 
> Was this a public API?

Not published; we only really used it for the mobile apps - those were already updated.

**Review by ermshiperete — APPROVED** [review]

**Comment by mcdurdin** [comment]
> The prediction and correction toggle properties... weren't even available there in 16.0, so the Web test-page toggles have likely been dead for a little while. I went ahead and patched them up while I was at it.

I don't really understand this, can you elaborate?

**Comment by mcdurdin** [comment]
> Not published; we only really used it for the mobile apps - those were already updated.

And ... Keyman Developer, clearly.

File: web/src/engine/package-cache/src/modelCache.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ export default class ModelManager {
     return this.languageModelMap[lgCode];
   }
 
-  // Accessible publicly as keyman.modelManager.register(model: ModelSpec)
+  // Accessible publicly as keyman.modelCache.register(model: ModelSpec)
```
Comment (mcdurdin) [inline web/src/engine/package-cache/src/modelCache.ts:14]: I see in test.js above you changed the `keyman.modelManager.register` reference to `keyman.addModel`, but here you refer to `keyman.modelCache.register`. Which should we be using?

File: web/src/test/manual/web/issue3701/index.html [hunk 2]
```diff
@@ -54,16 +54,16 @@
       });
 
       function togglePredictions() {
-        let newVal = !keyman.modelManager.mayPredict;
-        keyman.osk.banner.setOptions({'mayPredict': newVal});
+        let newVal = !keyman.core.languageProcessor.mayPredict;
+        keyman.core.languageProcessor.mayPredict = newVal;
```
Comment (mcdurdin) [inline web/src/test/manual/web/issue3701/index.html:58]: Is this a public API? Should it be?

File: web/src/engine/package-cache/src/modelCache.ts (hunk 1 above)
Comment (jahorton) [reply web/src/engine/package-cache/src/modelCache.ts:14]: I think the comment here was originally written before `addModel` was defined; that's the better one to use & make public, really.

File: web/src/test/manual/web/issue3701/index.html (hunk 2 above)
Comment (jahorton) [reply web/src/test/manual/web/issue3701/index.html:58]: I think it'd be wise to workshop a better API for that.  It's what we have for now, though.

Comment (mcdurdin) [reply web/src/test/manual/web/issue3701/index.html:58]: Agreed. For a future version then.

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_PREDICTIVE_TOGGLES (PASSED):** Tested with "Prediction - robust testing" for KeymanWeb on a mobile device (Android 13 - Samsung Galaxy A23 5g) and here is my observation: 1. Verified that the predictive-text banner was not visible after clicking the 'Disable predictions' button. 2. Also, verified that the predictive-text banner was visible after clicking the 'Enable predictions' button. Verified that the toggles have the indicated effect. 

..Disable predictions

   


..Enable predictions

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_DEV_HOST_MODELS (FAILED):** 1. Installed Keyman Developer 17.0.236-alpha-test-10328 on Windows 11 OS (VM). 2. Opened sil_euro_latin keyboard project. 3. Compiled the project. 4. Clicked the Build tab. Opened the test-host page. 5. Set a mobile test device configuration. 6. Noticed that the predictive text banner was missing in the OSK. 

   


..Verified that the lexical model loads properly

**Comment by jahorton** [comment]
> # Test Results
> 
>     * **TEST_DEV_HOST_MODELS (FAILED):** 1. Installed Keyman Developer 17.0.236-alpha-test-10328 on Windows 11 OS (VM). 2. Opened sil_euro_latin keyboard project. 3. Compiled the project. 4. Clicked the Build tab. Opened the test-host page. 5. Set a mobile test device configuration. 6. Noticed that the predictive text banner was missing in the OSK.
>        
> 
> 
> ..Verified that the lexical model loads properly
> 

@bharanidharanj  - I take it that the test failed when you had only loaded the keyboard, but passed once you also loaded the model for testing?

The test instructions did say to load _both_... because you _need both_ in order for the banner to appear.

**Comment by bharanidharanj** [comment]
> > # Test Results
> > ```
> > * **TEST_DEV_HOST_MODELS (FAILED):** 1. Installed Keyman Developer 17.0.236-alpha-test-10328 on Windows 11 OS (VM). 2. Opened sil_euro_latin keyboard project. 3. Compiled the project. 4. Clicked the Build tab. Opened the test-host page. 5. Set a mobile test device configuration. 6. Noticed that the predictive text banner was missing in the OSK.
> >    
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ..Verified that the lexical model loads properly
> > 
> 
> @bharanidharanj - I take it that the test failed when you had only loaded the keyboard, but passed once you also loaded the model for testing?
> 
> The test instructions did say to load _both_... because you _need both_ in order for the banner to appear.

@jahorton Yes, Josh. If I loaded only the keyboard, the predictive text did not appear on the OSK. If I load both Keyboard and Model then the banner appear on the OSK. Sorry, I have little bit confused with the testing instructions. It would be better If you reassign it for testing. Thanks.

**Comment by jahorton** [comment]
Per @bharanidharanj 's response,

TEST_DEV_HOST_MODELS: PASSED

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.243-alpha](https://keyman.com/downloads/releases/alpha/17.0.243)

**Pull request merged**